src/Database/Connection.php

```diff
@@ -278,44 +280,74 @@ public function run(Query $query): StatementInterface
         return $this->getDisconnectRetry()->run(fn () => $this->_driver->run($query));
     }
 
+    /**
+     * Get query factory instance.
+     *
+     * @return \Cake\Database\QueryFactory
+     */
+    public function queryFactory(): QueryFactory
+    {
+        return $this->queryFactory ??= new QueryFactory($this);
+    }
+
     /**
      * Create a new SelectQuery instance for this connection.
      *
+     * @param \Cake\Database\ExpressionInterface|\Closure|array|string|float|int $fields Fields/columns list for the query.
+     * @param array|string $tables List of tables to query.
+     * @param array<string, string> $types Associative array containing the types to be used for casting.
      * @return \Cake\Database\Query\SelectQuery
      */
-    public function newSelectQuery(): SelectQuery
-    {
-        return new SelectQuery($this);
+    public function newSelectQuery(
+        ExpressionInterface|Closure|array|string|float|int $fields = [],
+        array|string $tables = [],
+        array $types = []
+    ): SelectQuery {
+        return $this->queryFactory()->select($fields, $tables, $types);
     }
 
     /**
      * Create a new InsertQuery instance for this connection.
      *
+     * @param string|null $table The table to insert rows into.
+     * @param array $values Associative array of column => value to be inserted.
```

Comment:
Does this just represent a single row or can it be an array of VALUES()? It sounds like only supports a single row with a subset of column names.

Reply:
Yes it only supports a single row. I copied the API from the current `Connection::insert()` method.

Reply:
We can change the method to have separate $columns and $values arguments.

Reply:
Do we need a high-level API on QueryFactory to bulk insert records? The query builder already has methods for doing those operations.

Reply:
> Do we need a high-level API on QueryFactory to bulk insert records?

Nope, I prefer keeping the `QueryFactory` simple.